elixir-move.el

```diff
@@ -0,0 +1,291 @@
+          (when (< 0 (abs (skip-chars-backward " \t\r\n\f")))
+            (setq done t))
+          (back-to-indentation)
+          (elixir-beginning-of-statement orig done limit))
+         ;; after end of statement
+         ((and (not done) (eq (char-before) ?\;))
+          (skip-chars-backward ";")
+          (elixir-beginning-of-statement orig done limit))
+         ;; travel until indentation or semicolon
+         ((and (not done)
+               (elixir--skip-to-semicolon-backward
+                (save-excursion (back-to-indentation)(point))))
+          (elixir-beginning-of-statement orig done limit))
+         ;; at current indent
+         ((and (not done) (not (eq 0 (skip-chars-backward " \t\r\n\f"))))
+          (elixir-beginning-of-statement orig done limit)))
+        ;; return nil when before comment
+        (unless (and (looking-at "[ \t]*#") (looking-back "^[ \t]*"))
+          (when (< (point) orig)(setq erg (point))))
+        (when (and elixir-verbose-p (interactive-p)) (message "%s" erg))
+        erg))))
+
+(defalias 'elixir-beginning-of-statement-lc 'elixir-beginning-of-statement-bol)
+(defun elixir-beginning-of-statement-bol (&optional indent)
+  "Goto beginning of line where statement starts.
+  Returns position reached, if successful, nil otherwise.
+
+See also `elixir-up-statement': up from current definition to
+next beginning of statement above. "
+  (interactive)
+  (let* ((indent (or indent (when (eq 'elixir-end-of-statement-bol (car
+  elixir-bol-forms-last-indent))(cdr elixir-bol-forms-last-indent))))
+	 (orig (point))
+         erg)
+    (if indent
+        (while (and
+                (setq erg (elixir-beginning-of-statement))
+                (< indent (current-indentation))
+                (not (bobp))))
+      (setq erg (elixir-beginning-of-statement)))
+    ;; reset
+    (setq elixir-bol-forms-last-indent nil)
+    (beginning-of-line)
+    (and (< (point) orig) (setq erg (point)))
+    (when (interactive-p) (message "%s" erg))
+    erg))
+
+(defalias 'elixir-forward-statement 'elixir-end-of-statement)
+(defalias 'elixir-previous-statement 'elixir-end-of-statement)
+(defalias 'elixir-statement-forward 'elixir-end-of-statement)
+(defun elixir-end-of-statement (&optional orig done repeat)
+  "Go to the last char of current statement.
+
+Optional argument REPEAT, the number of loops done already,
+is checked for elixir-max-specpdl-size error.
+Avoid eternal loops due to missing string delimters etc. "
+  (interactive)
+  (unless (eobp)
+    (let ((repeat (or (and repeat (1+ repeat)) 0))
+          (orig (or orig (point)))
+          erg pos last
+          ;; use by scan-lists
+          parse-sexp-ignore-comments
+          forward-sexp-function
+          stringchar stm pps err)
+      (unless done (elixir--skip-to-comment-or-semicolon))
+      (setq pps (parse-partial-sexp (point-min) (point)))
+      ;; (origline (or origline (elixir-count-lines)))
```

Comment:
On 17.03.2015 21:24, Matt DeBoard wrote:
>> +
>> +Optional argument REPEAT, the number of loops done already,
>> +is checked for elixir-max-specpdl-size error.
>> +Avoid eternal loops due to missing string delimters etc. "
>> +  (interactive)
>> +  (unless (eobp)
>> +    (let ((repeat (or (and repeat (1+ repeat)) 0))
>> +          (orig (or orig (point)))
>> +          erg pos last
>> +          ;; use by scan-lists
>> +          parse-sexp-ignore-comments
>> +          forward-sexp-function
>> +          stringchar stm pps err)
>> +      (unless done (elixir--skip-to-comment-or-semicolon))
>> +      (setq pps (parse-partial-sexp (point-min) (point)))
>> +      ;; (origline (or origline (elixir-count-lines)))
>
> Can this be deleted?
>

Might remove these both files, the navigation stuff. Okay?